### ISSUE TITLE
CEserver and Client 7.5 stuck after opening process

### ISSUE THREAD

**Issue by dbbnsyg001** [cheat-engine/cheat-engine#2690]
First of all, please forgive my poor English and program analysis skills.
### CEServer : Github release version 7.5  and build with  Android NDK r21

When I use the 7.5 ceserver compiled **by myself,** I have used the forwarding of the 52736 port, and the client will **get stuck** after opening the process,
However, this situation does not occur when I use CEserver downloaded **from the official website**.

I found two commands, **java_OpenProcessAfterwards** and **SymbolLoaderThread downloaded** from the official website will immediately return **[Error: recv returned 0]** and  **[has disconnected]**

The two commands of the program I compiled will not return the recv error. I guess it is stuck in the recv() function. 

Dear author, what is the role of these two commands in the program?

### This is the official download of ceserver
```
Creating module list for process 16730
The opened process is 64-bit
processhandle=1
CMD_GETARCHITECTURE
(1)=3
Creating module list for process 16730
CMD_SET_CONNECTION_NAME
This thread is called Unnamed
accept=6
CMD_SET_CONNECTION_NAME
This thread is called java_OpenProcessAfterwards
CMD_SET_CONNECTION_NAME
This thread is called java_OpenProcessAfterwards
handleGetOptions
accept=8
CMD_SET_CONNECTION_NAME
This thread is called SymbolLoaderThread
CMD_SET_CONNECTION_NAME
This thread is called SymbolLoaderThread
CMD_SET_CONNECTION_NAME
This thread is called Unnamed
java_OpenProcessAfterwards: Error: recv returned 0
java_OpenProcessAfterwards has disconnected
accept=9
```

### This is the release version 7.5 of ceserver
```
Creating module list for process 8093
The opened process is 64-bit
processhandle=1
CMD_GETARCHITECTURE
(1)=3
Creating module list for process 8093
CMD_SET_CONNECTION_NAME
This thread is called Unnamed
accept=6
CMD_SET_CONNECTION_NAME
This thread is called java_OpenProcessAfterwards
CMD_SET_CONNECTION_NAME
This thread is called java_OpenProcessAfterwards
handleGetOptions
accept=8
CMD_SET_CONNECTION_NAME
This thread is called SymbolLoaderThread
CMD_SET_CONNECTION_NAME
This thread is called SymbolLoaderThread
```

**Comment by dbbnsyg001** [comment]
Ok I will try again with latest code

> with regards of the differences of the release 7.5 server on github and download i'm not sure. I may have commited the updates of ceserver after marking for release, so you may have to get the sourcecode from a few days later
> 
> but i recommend you get both the most recent cheat engine client and server from github and compile that. It works better

**Comment by dbbnsyg001** [comment]
I tried again and still have the same problem，but I narrowed down the error.

I found that as long as I compiled the ceserver myself, it would get stuck when the process was opened.

I downloaded the latest source code from github(2023-09-19). Use NDK r21b to compile the Android version  in the ndk-build\EXECUTABLE\jni directory.(Execute the command ***ndk-build***)

When I push to run on an Android device, when the windows client of CE7.5 opens the process, the windows client gets stuck.
